### PR TITLE
[DateInterval] Fix `func contains(_:) -> Bool`.

### PR DESCRIPTION
**Pull request by YOCKOW**
Resolves https://github.com/apple/swift-corelibs-foundation/issues/4588

### CODE REVIEW

**Comment by YOCKOW** [comment]
@swift-ci Please test

**Comment by YOCKOW** [comment]
@millenomi 
May I ping to request review from you?

File: Tests/Foundation/Tests/TestDateInterval.swift [hunk 1]
```diff
@@ -118,14 +118,25 @@ class TestDateInterval: XCTestCase {
...
+
+        __assert(interval: .init(start: .distantPast, end: .distantFuture), date: date0, expectedResult: true)
+        __assert(interval: .init(start: .distantPast, end: date1), date: date0, expectedResult: true)
+        __assert(interval: .init(start: .distantPast, end: date1), date: date2, expectedResult: false)
+        __assert(interval: .init(start: date0, end: date4), date: date2, expectedResult: true)
+        __assert(interval: .init(start: date1, duration: 60 * 45), date: date3, expectedResult: true)
+        __assert(interval: .init(start: date1, duration: 60 * 45), date: date4, expectedResult: false)
+        __assert(interval: .init(start: date2, end: .distantFuture), date: date2, expectedResult: true)
+        __assert(interval: .init(start: date2, end: .distantFuture), date: date0, expectedResult: false)
     }
```
[the remark this answers is not in the document]
Comment (YOCKOW) [reply Tests/Foundation/Tests/TestDateInterval.swift:140]: Sorry, but that was my habit.😅
I'll fix them.
Thank you for your review.

**Comment by YOCKOW** [comment]
@swift-ci Please test

**Comment by YOCKOW** [comment]
@parkera
Thank you for your review. Tests are fixed and passed now.

**Review by parkera — APPROVED** [review]

**Pull request merged**